Audit output-functions.md

MizuhoOkimoto opened the pull request:
Hi, this fixes #87 

This PR is an audit of output-functions.md file. I mainly added and fixed the below.
- Fix typos
- Remove extra spaces
- Unify markdown syntax
- Improve readability on the markdown file
- Check display (desktop/mobile/dark and light mode)
- Run the page through Lighthouse and Web Hint
- Added alt text to images
- Fix Frontmatter for the page to include proper id, title, and slug
- Remove extra indentation from code blocks

If I need to improve, please let me know! Thank you so much.

lmpham1 commented:
https://github.com/MizuhoOkimoto/IPC144/blob/08295546705c7dcde95516f7897f516f17aa4c01/docs/D-Modularity/output-functions.md?plain=1#L170

The "ASCII collating sequence" part at the end of this line can be linked to the [`ascii-collating-sequence.md` page](https://github.com/MizuhoOkimoto/IPC144/blob/issue-87/docs/Resources-Appendices/ascii-collating-sequence.md)

joshuali7536 commented on docs/D-Modularity/output-functions.md:
````diff
@@ -71,28 +75,28 @@ a
 ```
 
-***format*** is a set of characters enclosed in double-quotes that may consist of any combination of plain characters and conversion specifiers.  The function sends the plain characters as is to the buffer and uses the conversion specifiers to translate each value passed as an argument in the function call.  The ellipsis indicates that the number of arguments can vary.  Each conversion specifier  ...
+***format*** is a set of characters enclosed in double-quotes that may consist of any combination of plain characters and conversion specifiers. The function sends the plain characters as is to the buffer and uses the conversion specifiers to translate each value passed as an argument in the function call.  The ellipsis indicates that the number of arguments can vary.  Each conversion specifier c ...
````
Since you removed a bunch of the previous extra spaces, there are a few here that you missed. " The ellipsis" and " Each conversion" are the two I see.

joshuali7536 reviewed (commented):
Everything else looks good. I don't see any typos, or formatting errors (other than extra spaces I mentioned). The page looks good in light, dark, and mobile modes. Lighthouse and Hints didn't have any issues with the markdown part of the page. The alt text for images are clear and concise.

tcvan0707 commented on docs/D-Modularity/output-functions.md:
```diff
@@ -1,6 +1,10 @@
 sidebar_position: 5
+title: Output Functions
+id: output-functions
+slug: /modularity/output-functions
```
I think you should add description below the `slug` as well to let others know about the content in this page

tcvan0707 commented on docs/D-Modularity/output-functions.md:
```diff
@@ -11,17 +15,17 @@ After reading this section, you will be able to:
+The adequate provision of a user interface is an important aspect of software development: an interface that consists of user-friendly input and user-friendly output. The output facilities of a programming language convert the data in memory into a stream of characters that is read by the user. The stdio module of the C language provides such facilities.
 
-This chapter describes two functions in the stdio module that provide formatted and unformatted buffered support for streaming output data to the user and demonstrates in detail how to format output for a user friendly interface. 
+This chapter describes two functions in the stdio module that provide formatted and unformatted buffered support for streaming output data to the user and demonstrates in detail how to format output for a user-friendly interface.
```
You can change "stdio" into `stdio` for better visibility.

humphd reviewed (changes requested):
Let's get the changes to add backticks to stdio as suggested, so we can merge this

MizuhoOkimoto commented:
Hi @humphd! Thank you for reviewing! I added backticks for `stdio`, description, a hyperlink to the ASCII collating sequence page, and removed extra spaces. Please let me know if I need to change anything else 🐞⭐
